frb_codegen/src/library/codegen/parser/mir/parser/ty/concrete.rs

```diff
@@ -77,11 +77,28 @@ impl TypeParserWithContext<'_, '_, '_> {
                 Delegate(MirTypeDelegate::Map(MirTypeDelegateMap {
                     key: Box::new(key.clone()),
                     value: Box::new(value.clone()),
+                    hasher: None,
+                    element_delegate: self.create_mir_record(vec![key, value]),
+                }))
+            },
+            ("HashMap", [key, value, hasher]) => {
+                let key  = self.parse_type(key)?;
+                let value  = self.parse_type(value)?;
+                let hasher  = self.parse_type(hasher)?;
+                Delegate(MirTypeDelegate::Map(MirTypeDelegateMap {
+                    key: Box::new(key.clone()),
+                    value: Box::new(value.clone()),
+                    hasher: Some(Box::new(hasher)),
```

Comment:
nit: maybe we can extract a function, say, `fn parse_mir_hash_map(key, value, hasher: Optional<...>)`, and call it in this two branches

(same for hashset)